src/Serialization/upgrades/0.11.3.jl

```diff
@@ -0,0 +1,118 @@
+                    upgraded_dict[key] = upgrade_0_11_3(s, value)
+                else
+                    values = []
+                    for v in value
+                        if v isa String
+                            push!(values, v)
+                        else
+                            push!(values, upgrade_0_11_3(s, v))
+                        end
+                    end
+                    upgraded_dict[key] = values
+                end
+            end
+            return upgraded_dict
+        end
+
+        if dict[:type] == string(backref_sym)
+            backref = s.objs[UUID(dict[:id])]
+
+            # data no longer uses backref and is part of the update
+            backref isa String && return backref
+
+            return dict
+        end
+
+        # this handles types like FlintRationalField that have no associated data
+        !haskey(dict, :data) && return dict
+
+        # apply upgrade to vector entries
+        # we only do this here since upgrade_0_11_13 is only defined for dicts
+        # we could make this it's own function outside of this script
+        if dict[:type] == "Vector"
+            upgraded_vector = []
+            entry_type = nothing
+
+            for entry in dict[:data][:vector]
+                result = upgrade_0_11_3(s, entry)
+
+                # store values in state that are vector entries that aren't backrefs
+                if entry[:type] != string(backref_sym)
+                    if haskey(entry, :id)
+                        s.objs[UUID(entry[:id])] = result
+                    end
+                    entry_type = entry[:type]
+                end
+
+                push!(upgraded_vector, result)
+            end
+
+            # no update at this level
+            upgraded_vector[1] isa Dict && return Dict(
+                :type => "Vector",
+                :id => dict[:id],
+                :data => Dict(
+                    :vector => upgraded_vector
+                )
+            )
+
+            return Dict(
+                :type => "Vector",
+                :id => dict[:id],
+                :data => Dict(
+                    :vector => upgraded_vector,
+                    :entry_type => entry_type
+                )
+            )
+        end
+
+        U = decodeType(dict[:type])
+
+        # Upgrades fmpq serialization
+        if is_basic_serialization_type(U)
+            if U === fmpq
+                num = dict[:data][:num][:data]
+                den = dict[:data][:den][:data]
+                return "$num//$den"
+            else
+                return dict[:data]
+            end
+        end
+
+        upgraded_data = upgrade_0_11_3(s, dict[:data])
+        upgraded_dict = Dict(
+            :type => dict[:type],
+            :data => upgraded_data,
+            :id => dict[:id]
+        )
+
+        # adds updated object in case it is referenced somewhere
+        if haskey(dict, :id)
+            s.objs[UUID(dict[:id])] = upgraded_dict
```

Comment:
Do we also need do this for the code handling type `"Vector"` in lines 73-88?

Reply:
i've added objs to state if they have ids, before each return now